server/server.js

```diff
@@ -269,10 +270,18 @@ app.get('/genotet', function(req, res) {
 
     // Bed data queries
     case 'bed':
-      var file = req.query.file;
+      var file = req.query.fileName;
       var chr = req.query.chr;
-      var dir = bedPath + file + '_chr/' + file + '_' + chr;
-      data = bed.readBed(dir, req.query.xl, req.query.xr);
+      var dir = bedPath + file + '_chr/' + file + '_chr' + chr;
+      var xl = -Infinity, xr = Infinity;
+      if (req.query.hasOwnProperty('xl')) {
+        xl = req.query.xl;
+        xr = req.query.xr;
+      }
+      data = bed.readBed(dir, xl, xr);
```

Comment:
So you are not going to use bed.query.motifs() anymore but use bed.readBed_() directly? Why?